### DISCUSSION TITLE
HCL Digital Experience 9.5 (formerly Websphere Portal)

### DISCUSSION THREAD

**Discussion by hAzUpL8** [primefaces/community#1246] [PrimeFaces] [answered]
Hi

I ran into a problem navigating between pages on a portlet. So far we have never had similar problems with this combination of software!

Let me describe the situation:

When migrating existing portlet applications from the old version of 
_HCL Digital Experience Portal v8.5, java 1.8, JSF2.0 running on Apache MyFaces with Primefaces v7.0.3_
to the new version of 
_HCL Digital Experience Portal v9.5 CF216, java 1.8, JSF 2.2 running on Apache MyFaces-2 with Primefaces v13.0.1_
I encountered the following problem. When navigating between pages with the commandButton, if I use **h:commandButton**, it works, but if I replace it with **p:commandButton**, it doesn't.

My code looks like this
```
<h:form id="form">
     <p:autoUpdate/>
     <p:messages id="msgs" skipDetailIfEqualsSummary="true"/>

     <p:inputText id="ena" value="#{managedBean.name}" />
     <p:inputText id="dva" value="#{managedBean.lastName}" />
     <p:inputText id="tri" value="#{managedBean.age}" />

    <h:commandButton value="Test me 'h:'" action="View2.xhtml"/> <!-- this button works OK -->
    <p:commandButton value="Test me 'p:'" action="View2.xhtml"/> <!-- this one doesn't -->
</h:form>
```
Given that in my case the "age" field is an integer and if I enter a letter in this field both buttons validate correctly! But if the content is correct, **h:commandButton** navigates to the View2.xhtml page, but **p:commandButton** does not and stays on the same page.
There is no error in the logs, only the portal page completely crashes.

I don't know how to proceed further. What do you advise me?
So far I can't open a case on HCL because they will say that as far as they are concerned <h:commandButton> is working correctly!

regards Haris

**Comment by tandraschko** [comment, marked as answer]
i think you need to debug CommandButton/CommandButtonRenderer and also the same on MyFaces to get the difference
also important artifacts are NavigationHandler or the impl of it in MyFaces
i never heard of that container and we dont have much porlet users here

Also note that PrimeFaces 14 is not compatible with JSF 2.2 anymore, it requires 2.3+
